pkg/controllers/management/auth/manager.go

```diff
@@ -762,3 +762,18 @@ func buildRule(resource string, verbs map[string]bool) v1.PolicyRule {
 		APIGroups: []string{"*"},
 	}
 }
+
+func (m *manager) getBuiltinParentRole(roleTemplateName string) (*v3.RoleTemplate, error) {
+	roleTemplate, err := m.rtLister.Get("", roleTemplateName)
+	if err != nil {
+		return nil, err
+	}
+	if !roleTemplate.Builtin && len(roleTemplate.RoleTemplateNames) > 0 {
+		// get referenced roletemplate
+		for _, rt := range roleTemplate.RoleTemplateNames {
+			return m.getBuiltinParentRole(rt)
```

Comment:
q: what's the loop for, looks like we always return

Reply:
Getting parent roles for the roles that the current roletemplate is referencing recursively